lark/lexer.py

```diff
@@ -384,12 +397,25 @@ def _build_mres(self, terminals, max_size):
             terminals = terminals[max_size:]
         return mres
 
-    def match(self, text, pos):
+    def match(self, text, pos, *, end_pos=sys.maxsize):
         for mre in self._mres:
-            m = mre.match(text, pos)
+            m = mre.match(text, pos, end_pos)
             if m:
                 return m.group(0), m.lastgroup
 
+    def search(self, text, start_pos, end_pos):
+        best = None, float("inf")
+        for mre in self._mres:
+            mre: re.Pattern
+            m = mre.search(text, start_pos, end_pos)
+            if m:
+                if m.start() < best[1]:
+                    best = (m.group(0), m.lastgroup), m.start()
```

Comment:
won't it be simpler with `best = m`?

Reply:
A bit. But in my understanding, the idea behind `Scanner` is that it abstracts away from the details of the `re` implementation, so I still need to do all the accesses in this function.

Reply:
Yep, it's just slightly simpler logic inside the function. (i think)